Neos.Fusion/Classes/FusionObjects/AbstractArrayFusionObject.php

```diff
@@ -130,27 +146,65 @@ protected function evaluateNestedProperties(?string $defaultFusionPrototypeName
      *
      * This will ignore all properties defined in "@ignoreProperties" in Fusion
      *
-     * @see PositionalArraySorter
-     *
      * @return array an ordered list of key value pairs
      * @throws FusionException if the positional string has an unsupported format
+     * @see PositionalArraySorter
+     *
+     * @deprecated
+     * @see preparePropertyKeys()
      */
     protected function sortNestedProperties(): array
     {
-        $arraySorter = new PositionalArraySorter($this->properties, '__meta.position');
+        return $this->preparePropertyKeys($this->properties);
+    }
+
+    /**
+     * @param array $properties
+     * @return array<string> Fusion keys in this Array fusion object
+     * @throws FusionException
+     */
+    protected function preparePropertyKeys(array $properties): array
+    {
+        $maybeSortedFusionKeys = $this->shouldSortProperties() ? $this->applyPositionalArraySorterToProperties($properties) : array_keys($properties);
+        return $this->filterIgnoredProperties($maybeSortedFusionKeys);
+    }
+
+    /**
+     * @param array $properties
+     * @return array
+     * @throws FusionException
+     */
+    protected function applyPositionalArraySorterToProperties(array $properties): array
```

Comment:
i would consider making this private, so its cannot be used as api what do you say?
this is more of a general thing to use `private` more often - but probably just boils down to own taste.

Reply:
yep, thought about it, and _probably_ better...

Reply:
Ah, it cannot be private because we do call this from implementations further up the chain...